client/src/components/CountdownTimer.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useEffect, useRef } from 'react';
+
+const CountdownTimer: React.FC = () => {
+  const [timeLeft, setTimeLeft] = useState<number>(60);
+  const [isActive, setIsActive] = useState<boolean>(false);
+  const intervalRef = useRef<number | null>(null);
+
+  useEffect(() => {
+    if (isActive && timeLeft > 0) {
+      intervalRef.current = window.setInterval(() => {
+        setTimeLeft((prevTime) => prevTime - 1);
+      }, 1000);
+    } else if (timeLeft === 0 || !isActive) {
+      if (intervalRef.current) {
+        clearInterval(intervalRef.current);
+        intervalRef.current = null;
+      }
+    }
+
+    return () => {
+      if (intervalRef.current) {
+        clearInterval(intervalRef.current);
+      }
+    };
+  }, [isActive, timeLeft]);
+
+  const handleStart = () => {
+    setIsActive(true);
+  };
+
+  const handleReset = () => {
+    setIsActive(false);
+    setTimeLeft(60);
+  };
+
+  const handleStop = () => {
+    setIsActive(false);
+  };
+
+  const isCritical = timeLeft < 10;
```

Comment:
Add more checkpoints like this through the process:
One for 45 seconds, one for 30 and one for 20s
Choose the colors so that they convert from blue to red
Also try to ease in into the transition